File: app/src/main/java/vng/zalo/tdtai/zalo/zalo/views/lobby/fragments/contact_fragment/ContactFragmentAdapter.java
package vng.zalo.tdtai.zalo.zalo.views.lobby.fragments.contact_fragment;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;
import vng.zalo.tdtai.zalo.zalo.views.lobby.fragments.contact_fragment.tab_fragments.ContactSubFragment;
import vng.zalo.tdtai.zalo.zalo.views.lobby.fragments.contact_fragment.tab_fragments.OfficialAccountSubFragment;

public class ContactFragmentAdapter extends FragmentStatePagerAdapter {
    private Fragment officialAccountSubFragment;
    private Fragment contactSubFragment;

    ContactFragmentAdapter(FragmentManager fm) {
        super(fm);

        contactSubFragment = new ContactSubFragment();
        officialAccountSubFragment = new OfficialAccountSubFragment();
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        // getItem is called to instantiate the fragment for the given page.
        // Return a PlaceholderFragment (defined as a static inner class below).
        switch (position){
            case 1:
                return officialAccountSubFragment;
            default:
                return contactSubFragment;
        }
    }

    @Override
    public int getCount() {
        return 2;
    }
}

File: app/src/main/java/vng/zalo/tdtai/zalo/zalo/adapters/ContactFragmentAdapter.kt
package vng.zalo.tdtai.zalo.zalo.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import vng.zalo.tdtai.zalo.zalo.views.fragments.ContactSubFragment
import vng.zalo.tdtai.zalo.zalo.views.fragments.OfficialAccountSubFragment

class ContactFragmentAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
    private var officialAccountSubFragment: Fragment? = null
    private var contactSubFragment: Fragment? = null

    override fun getItem(position: Int): Fragment {
        return when (position) {
            1 -> {
                if (officialAccountSubFragment == null) {
                    officialAccountSubFragment = OfficialAccountSubFragment()
                }
                officialAccountSubFragment!!
            }
            else -> {
                if (contactSubFragment == null) {
                    contactSubFragment = ContactSubFragment()
                }
                contactSubFragment!!
            }
        }
    }

    override fun getCount(): Int {
        return 2
    }
}